### DISCUSSION TITLE
How to troubleshoot Dexcom upload

### DISCUSSION THREAD

**Discussion by erizzo** [NightscoutFoundation/xDrip#2015] [Q&A]
Using xDrip with Dexcom G6, I'm trying to enable uploading to Dexcom's servers (using my Dexcom account). I've entered my username and password (3 times, actually), but no data is showing up in, for example, Dexcom Clarity.
The xDrip UI for upload configuration is so rudimentary, there's no feedback about whether or not xDrip is able to connect to my account to upload data - nothing. How can I troubleshoot why the data isn't showing up?

**Comment by Navid200** [comment]
I don't believe you can upload to Clarity from xDrip.
You can upload to Dex share servers, which is something different than Clarity.

**Reply by erizzo** [reply]
So the Dexcom Share that xDrip supports seems to be only for "followers" who use the Dexcom Follow app.
Since xDrip doesn't support sharing data via Clarity (which is what a provider/doctor is most likely to use), what other mechanisms does xDrip support for sharing my data with my provider?

**Reply by jonprogers** [reply]
Personally I do all my reporting via either nightscout inbuilt reports or by using nightscout reporter at https://nightscout-reporter.zreptil.de/ 
My endo is always happy with both options.
I have also shared my nightscout URL with my healthcare team but suspect it is never used by them...

**Comment by erizzo** [comment]
Also, I tried using xDrip's "Invite Followers" feature, but every time I try to invite someone, xDrip just shows a toast saying "Failed to invite follower"
Is there a way to troubleshoot why it's failing?

**Reply by Navid200** [reply]
xDrip cannot upload to Clarity.  It's not xDrip's choice.

If you want to share using Dex share, after you set up the master, you can select "Dex Share Follower" as the hardware data source of the xDrip of the follower phone.

There are different ways you can provide reports to your doctor.
One is to create a very simple report:  https://navid200.github.io/xDrip/docs/Report.html
